src/app/store/teacher-subjects/teacher-subjects.reducer.ts

```diff
@@ -0,0 +1,22 @@
+import { Action, createReducer, on } from "@ngrx/store";
+import * as SubjectData from "./teacher-subjects.action";
+
+export interface State {
+  data: Array<Object>;
```

Comment:
this `Object` might be a reason why  here https://github.com/webui100/SchoolProject/pull/37/files#diff-63e2fbde51a7d0f49129c728d08de2ddR25 and in few other places you have to put ignore ))